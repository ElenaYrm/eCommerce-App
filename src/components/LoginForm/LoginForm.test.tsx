import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as reduxHooks from 'react-redux';
import { LoginForm } from './index';

jest.mock('react-redux');
jest.mock('../../constant/metaData', () => {});
jest.mock('../../services/sdk/auth/methods', () => {});

const useDispatchSpy = jest.spyOn(reduxHooks, 'useDispatch');

describe('Test LoginForm component: ', () => {
  test('should create LoginForm', () => {
    useDispatchSpy.mockReturnValue(jest.fn());
    const component = render(<LoginForm />);
    expect(component).toMatchSnapshot();
  });

  test('should dispatch action on submit', async () => {
    const mockDispatchFn = jest.fn();
    useDispatchSpy.mockReturnValue(mockDispatchFn);

    render(<LoginForm />);

    const submitBtn = screen.getByText('Login ( ^Ï‰^)');
    expect(submitBtn).toBeInTheDocument();
    await userEvent.click(submitBtn);
    expect(mockDispatchFn).not.toHaveBeenCalled();

    const email = screen.getByPlaceholderText('Email');
    expect(email).toBeInTheDocument();
    await userEvent.type(email, 'test@test.com');

    const password = screen.getByPlaceholderText('Password');
    expect(email).toBeInTheDocument();
    await userEvent.type(password, 'testTest1!');

    await userEvent.click(submitBtn);
    expect(mockDispatchFn).toHaveBeenCalled();
  });
});
