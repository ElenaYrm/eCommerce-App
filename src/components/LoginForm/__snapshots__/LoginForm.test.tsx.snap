// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginForm component:  should create LoginForm 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
        novalidate=""
      >
        <div
          class="form__input"
        >
          <label>
            <span
              class="visually-hidden"
            >
              email
            </span>
            <input
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="form__input"
        >
          <label>
            <span
              class="visually-hidden"
            >
              password
            </span>
            <input
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <button
              class="button__toggle"
              type="button"
            >
              Show
            </button>
          </label>
        </div>
        <button
          class="button__primary"
          type="submit"
        >
          Login ( ^ω^)
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
      novalidate=""
    >
      <div
        class="form__input"
      >
        <label>
          <span
            class="visually-hidden"
          >
            email
          </span>
          <input
            name="email"
            placeholder="Email"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="form__input"
      >
        <label>
          <span
            class="visually-hidden"
          >
            password
          </span>
          <input
            name="password"
            placeholder="Password"
            type="password"
            value=""
          />
          <button
            class="button__toggle"
            type="button"
          >
            Show
          </button>
        </label>
      </div>
      <button
        class="button__primary"
        type="submit"
      >
        Login ( ^ω^)
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
